heka/sandbox/decoders/http_edge_decoder.lua

```diff
@@ -173,6 +174,19 @@ function process_message()
     main_msg.Fields.geoCountry = get_geo_country(xff, remote_addr)
 
     -- Send new message along.
-    inject_message(main_msg)
-    return 0
+    local main_status, main_err = pcall(inject_message, main_msg)
+    local error_message = ""
+    local return_code = 0
+    if not landfill_status then
```

Comment:
Yeah, that is not the way it works. The failure will discard all messages so even in main was successful it will never see the light of day.